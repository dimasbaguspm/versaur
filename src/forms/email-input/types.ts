import type { TextInputProps } from '../text-input/types'

/**
 * Props for the EmailInput component
 * Extends TextInputProps and enforces type="email" for HTML input standards
 */
export type EmailInputProps = Omit<TextInputProps, 'type'>
