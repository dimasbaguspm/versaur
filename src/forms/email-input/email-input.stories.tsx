/**
 * EmailInput stories for Versaur UI
 *
 * Demonstrates usage and variations of the EmailInput component
 */
import type { Meta, StoryObj } from '@storybook/react'
import { EmailInput } from './email-input'

const meta: Meta<typeof EmailInput> = {
  title: 'Forms/EmailInput',
  component: EmailInput,
}
export default meta

type Story = StoryObj<typeof EmailInput>

export const Basic: Story = {
  args: {
    label: 'Email',
    placeholder: 'your@email.com',
    required: true,
  },
}

export const WithHelperText: Story = {
  args: {
    label: 'Email',
    helperText: 'We will never share your email.',
    placeholder: 'your@email.com',
  },
}

export const ErrorState: Story = {
  args: {
    label: 'Email',
    error: 'Invalid email address',
    required: true,
    placeholder: 'your@email.com',
  },
}
