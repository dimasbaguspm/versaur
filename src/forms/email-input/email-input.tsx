import { forwardRef } from 'react'
import { TextInput } from '../text-input/text-input'
import type { EmailInputProps } from './types'
import { Icon } from '@/primitive/icon'
import { MailIcon } from 'lucide-react'

/**
 * EmailInput component for Versaur UI
 *
 * A wrapper around TextInput that enforces type="email" and follows HTML input standards
 * Supports all TextInput props except type, which is always "email"
 */
export const EmailInput = forwardRef<HTMLInputElement, EmailInputProps>(
  (props, ref) => {
    return (
      <TextInput
        {...props}
        ref={ref}
        type='email'
        autoComplete='email'
        inputMode='email'
        leftContent={<Icon as={MailIcon} color='inherit' size='sm' />}
      />
    )
  }
)
