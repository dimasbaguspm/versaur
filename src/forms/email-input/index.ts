export { EmailInput } from './email-input'
export type { EmailInputProps } from './types'
