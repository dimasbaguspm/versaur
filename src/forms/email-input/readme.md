# EmailInput

A wrapper component for `TextInput` that enforces HTML email input standards.

## Usage
```tsx
import { EmailInput } from '@/forms/email-input'

<EmailInput label="Email" required />
```

## Props
- Inherits all `TextInput` props except `type` (always set to "email")
- Adds `autoComplete="email"` and `inputMode="email"` for best browser support
