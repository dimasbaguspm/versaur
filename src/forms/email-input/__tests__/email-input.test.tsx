import { render } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import * as stories from '../email-input.stories'

describe('EmailInput', () => {
  const { Basic, WithHelperText, ErrorState } = composeStories(stories)

  it('renders correctly (snapshot)', () => {
    const { asFragment } = render(<Basic />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with helper text', () => {
    const { getByText } = render(<WithHelperText />)
    expect(getByText('We will never share your email.')).toBeInTheDocument()
  })

  it('shows error state', () => {
    const { getByText } = render(<ErrorState />)
    expect(getByText('Invalid email address')).toBeInTheDocument()
  })

  it('uses type="email" and proper attributes', () => {
    const { getByLabelText } = render(<Basic />)
    const input = getByLabelText('Email') as HTMLInputElement
    expect(input.type).toBe('email')
    expect(input.autocomplete).toBe('email')
    expect(input.getAttribute('inputmode')).toBe('email')
  })
})
