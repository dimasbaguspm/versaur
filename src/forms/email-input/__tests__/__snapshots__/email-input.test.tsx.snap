// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmailInput > renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-sm font-medium text-foreground mb-2"
      for="«r0»"
    >
      Email
    </label>
    <div
      class="relative w-full"
    >
      <input
        aria-invalid="false"
        autocomplete="email"
        class="block w-full rounded-md border bg-white transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none disabled:bg-gray-50 border-primary/30 text-foreground focus:border-primary focus:ring-primary/20 pl-3 pr-9 h-10"
        id="«r0»"
        inputmode="email"
        placeholder="your@email.com"
        required=""
        type="email"
      />
      <span
        class="absolute right-2.5 top-0 bottom-0 pointer-events-none text-gray-500 flex items-center justify-center w-5"
        data-testid="right-content"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-mail inline-flex items-center justify-center w-4 h-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
          />
          <rect
            height="16"
            rx="2"
            width="20"
            x="2"
            y="4"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
